FreeRTOS/FreeRTOS-Plus-TCP
# Add integer overflow checks to buffer allocation APIs

Description
-----------
This PR updates the buffer allocation functions with checks to verify that integer additions wont lead to an overflow. 

Test Steps
-----------
Validated by running the HW demos.
Verified by creating scenarios leading to overflow.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
